src/stactools/sentinel2/stac.py

```diff
@@ -311,7 +315,25 @@ def set_asset_properties(_asset: pystac.Asset, _band_gsd: Optional[int] = None):
         asset_eo = EOExtension.ext(asset)
         asset_eo.bands = [band_from_band_id(band_id)]
         set_asset_properties(asset, band_gsd)
-        return asset_key, asset
+
+        if processing_baseline < "04.00":
+            scale = 0.0001  # DN = 10000 * REFLECTANCE
+            offset = 0
+        else:  # todo: pull this from metadata
+            scale = 0.0001
+            offset = -1000
```

Comment:
The offset is -1000 DNs, but the scale/offset follows the GDAL convention which is:

new_val = scale*val + offset

So the offset needs to be in final "units" of reflectance, i.e., multiple the DN offset by the gain, see #44

Reply:
thx, I hadn't seen that ticket. will update.